### ISSUE TITLE
[Proposal] Replacing eslint-plugin-react with eslint-react

### ISSUE THREAD

**Issue by Talent30** [AndreaPontrandolfo/sheriff#189]
eslint-react is a modern version of eslint-plugin-react
https://eslint-react.xyz/docs/faq
https://github.com/Rel1cx/eslint-react/issues/85

I will send a PR shortly if @AndreaPontrandolfo likes this idea.

**Comment by AndreaPontrandolfo** [comment]
Hey @Talent30 !

Sheriff is already using `@eslint-react/eslint-plugin`, but just for a couple of rules that i find useful and are not covered by `eslint-plugin-react`.
That being said, at this stage i don't consider `@eslint-react/eslint-plugin` a replacement for `eslint-plugin-react`. It's a very interesting project but it is still young and is still growing, it will take a while to reach the level of maturity of a project like `eslint-plugin-react`.

In summary, Sheriff is already employing `@eslint-react/eslint-plugin` so no PR is needed, thank you anyway for the report.
I'm gonna close this now.

**Comment by lishaduck** [comment]
@AndreaPontrandolfo, maybe move this to a discussion? I think that at some point, it'd be nice to swap over to @eslint-react, but stick with both for now, given that it's a maturity/compatibility issue. Just my 2¢.

**Comment by AndreaPontrandolfo** [comment]
So, one of the issues with this, is that `@eslint-react/eslint-plugin` isn't a drop-in replacement for `eslint-plugin-react`, as outlined [here](https://eslint-react.xyz/docs/faq#is-there-anything-to-note-about-migrating-from-eslint-plugin-react-to-eslint-plugin-react-x). 
So transitioning wouldn't be trivial, and it's questionable whether it would be an objective improvement or not.

**Comment by lishaduck** [comment]
> So, one of the issues with this, is that `@eslint-react/eslint-plugin` isn't a drop-in replacement for `eslint-plugin-react`, as outlined [here](https://eslint-react.xyz/docs/faq#is-there-anything-to-note-about-migrating-from-eslint-plugin-react-to-eslint-plugin-react-x).  So transitioning wouldn't be trivial.

Yeah, so maybe wait for v30? (Maybe make majors that're multiples of majorly major? I personally like [RedwoodJS's Epochs concept](https://tom.preston-werner.com/2022/05/23/major-version-numbers-are-not-sacred.html).)

>It's questionable whether it would be an objective improvement or not.

Maybe not yet, but at some point it will likely mature, and then it's a more flexible, feature-complete, and more maintained plugin.

---

Anyway, I still vote this becomes an "ideas" discussion.
